charts/kubearchive/templates/api_server/api_server.yaml

```diff
@@ -24,31 +24,14 @@ spec:
             - name: tls-secret
               readOnly: true
               mountPath: /etc/kubearchive/ssl/
+          envFrom:
+            - secretRef:
+                name: {{ .Values.database.secret }}
```

Comment:
Why does the API server need the secret but no other database information?

Reply:
it needs all the database information, the api server, the same as the sink has all these variables:
* POSTGRES_USER
* POSTGRES_PASSWORD
* POSTGRES_DATABASE
* POSTGRES_URL
* POSTGRES_PORT

The first four are being mounted through the Kubernetes Secret in this line.
The change I did here is to simplify, when `envFrom.secretRef` is used it mounts all the values from a secret as environment variables so you don't need to specify them one by one.

Reply:
In that case, I suggest moving POSTGRES_PORT to the secret as well.  It looks out of place and is confusing to only see the port specified explicitly here.

Reply:
`envFrom` unpacks the keys of the secret into the environment. As we were unpacking every key into the environment, this makes it more compact. So it only needs the secret name.

Reply:
does it make sense to have the port in a secret with the value encoded? Isn't better to have it in clear text?

I can add a comment to prevent confusion

Reply:
But how is the port different than the user, URL, and database?  Why are those all encoded?  The only one that really needs encoding is the password.

But I'm all for keeping them together.

This also could help us making the database a true integration.  You'd (potentially) only need to change the secret to support a new database.

Reply:
I think it would make sense to have it as a secret as we already have the URL. They are not really secrets, but obfuscating them is somewhat useful. Same way you would like the SSH port to be secret if you changed it from the default one.

Reply:
ok, I'll make the change then